### ISSUE TITLE
Problem with JDBC-Tomcat and Micrometer

### ISSUE THREAD

**Issue by ivangfr** [micronaut-projects/micronaut-data#199]
Hi,

When using `jdbc-tomcat` and `micrometer`
```
implementation "io.micronaut.configuration:micronaut-jdbc-tomcat"
...
implementation "io.micronaut.configuration:micronaut-micrometer-core"
implementation "io.micronaut.configuration:micronaut-micrometer-registry-prometheus"
```

And then building the native image of the application, while trying to run the app I am getting the following exception
```
[main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [org.hibernate.SessionFactory] could not be loaded: Error instantiating bean of type  [io.micrometer.core.instrument.binder.MeterBinder]

Message: org.springframework.jdbc.datasource.DelegatingDataSource
Path Taken: CompositeMeterRegistry.compositeMeterRegistry(List registries,[List configurers]) --> MeterRegistryConfigurer.meterRegistryConfigurer([Collection binders],Collection filters) --> MeterBinder.dataSourceMeterBinder(String dataSourceName,[DataSourcePoolMetadata dataSourcePoolMetadata])
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [org.hibernate.SessionFactory] could not be loaded: Error instantiating bean of type  [io.micrometer.core.instrument.binder.MeterBinder]

Message: org.springframework.jdbc.datasource.DelegatingDataSource
Path Taken: CompositeMeterRegistry.compositeMeterRegistry(List registries,[List configurers]) --> MeterRegistryConfigurer.meterRegistryConfigurer([Collection binders],Collection filters) --> MeterBinder.dataSourceMeterBinder(String dataSourceName,[DataSourcePoolMetadata dataSourcePoolMetadata])
        at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1276)
        at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:236)
        at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2458)
        at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:202)
        at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:187)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:69)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:303)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:289)
        at com.mycompany.micronautbookapi.Application.main(Application.java:14)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micrometer.core.instrument.binder.MeterBinder]

Message: org.springframework.jdbc.datasource.DelegatingDataSource
Path Taken: CompositeMeterRegistry.compositeMeterRegistry(List registries,[List configurers]) --> MeterRegistryConfigurer.meterRegistryConfigurer([Collection binders],Collection filters) --> MeterBinder.dataSourceMeterBinder(String dataSourceName,[DataSourcePoolMetadata dataSourcePoolMetadata])
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1636)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2319)
        at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2001)
        at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:1975)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1094)
        at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1007)
        at io.micronaut.configuration.metrics.binder.datasource.$DataSourcePoolMetricsBinderFactory$DataSourceMeterBinderDefinition.doBuild(Unknown Source)
        at io.micronaut.context.AbstractParametrizedBeanDefinition.build(AbstractParametrizedBeanDefinition.java:117)
        at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:109)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1610)
        at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:2642)
        at io.micronaut.context.DefaultBeanContext.getBeansOfTypeInternal(DefaultBeanContext.java:2564)
        at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:923)
        at io.micronaut.context.AbstractBeanDefinition.lambda$getBeansOfTypeForConstructorArgument$10(AbstractBeanDefinition.java:1114)
        at io.micronaut.context.AbstractBeanDefinition.resolveBeanWithGenericsFromConstructorArgument(AbstractBeanDefinition.java:1752)
        at io.micronaut.context.AbstractBeanDefinition.getBeansOfTypeForConstructorArgument(AbstractBeanDefinition.java:1109)
        at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:984)
        at io.micronaut.configuration.metrics.micrometer.$MeterRegistryFactory$MeterRegistryConfigurerDefinition.build(Unknown Source)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1610)
        at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:2642)
        at io.micronaut.context.DefaultBeanContext.getBeansOfTypeInternal(DefaultBeanContext.java:2564)
        at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:923)
        at io.micronaut.context.AbstractBeanDefinition.lambda$getBeansOfTypeForConstructorArgument$10(AbstractBeanDefinition.java:1114)
        at io.micronaut.context.AbstractBeanDefinition.resolveBeanWithGenericsFromConstructorArgument(AbstractBeanDefinition.java:1752)
        at io.micronaut.context.AbstractBeanDefinition.getBeansOfTypeForConstructorArgument(AbstractBeanDefinition.java:1109)
        at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:984)
        at io.micronaut.configuration.metrics.micrometer.$MeterRegistryFactory$CompositeMeterRegistryDefinition.build(Unknown Source)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1610)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2319)
        at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2001)
        at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:1975)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:617)
        at io.micronaut.context.UnresolvedProvider.get(UnresolvedProvider.java:58)
        at io.micronaut.configuration.hibernate.jpa.metrics.HibernateMetricsBinder.onCreated(HibernateMetricsBinder.java:80)
        at io.micronaut.configuration.hibernate.jpa.metrics.HibernateMetricsBinder.onCreated(HibernateMetricsBinder.java:49)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1668)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2319)
        at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:1900)
        at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1274)
        ... 8 common frames omitted
Caused by: java.lang.NoClassDefFoundError: org.springframework.jdbc.datasource.DelegatingDataSource
        at io.micronaut.jdbc.spring.SpringDataSourceResolver.resolve(SpringDataSourceResolver.java:37)
        at io.micronaut.configuration.jdbc.tomcat.DatasourceFactory.tomcatPoolDataSourceMetadataProvider(DatasourceFactory.java:82)
        at io.micronaut.configuration.jdbc.tomcat.$DatasourceFactory$TomcatPoolDataSourceMetadataProvider1Definition.build(Unknown Source)
        at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:113)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1610)
        ... 46 common frames omitted
```

Best regards

**Comment by graemerocher** [comment]
This is resolved and fixed with https://github.com/micronaut-projects/micronaut-core/commit/6b146e4c4067257669438b0e8d20c775b2cb7915

It will be resolved in Micronaut 1.2.4 when that is released (today or tomorrow)

**Comment by ivangfr** [comment]
Thanks @graemerocher 👍
